apis/apiextensions/v1/composition_types.go

```diff
@@ -328,25 +446,79 @@ func (c *Patch) applyFromFieldPathPatch(from, to runtime.Object) error { // noli
 	if err != nil {
 		return err
 	}
-	out := in
-	for i, f := range c.Transforms {
-		if out, err = f.Transform(out); err != nil {
-			return errors.Wrapf(err, errFmtTransformAtIndex, i)
-		}
+
+	// Apply transform pipeline
+	out, err := c.applyTransforms(in)
+	if err != nil {
+		return err
 	}
 
-	if u, ok := to.(interface{ UnstructuredContent() map[string]interface{} }); ok {
-		return fieldpath.Pave(u.UnstructuredContent()).SetValue(*c.ToFieldPath, out)
+	return patchFieldValueToObject(*c.ToFieldPath, out, to)
+}
+
+// applyCombineFromVariablesPatch patches the "to" resource, taking a list of
+// input variables and combining them into a single output value.
+// The single output value may then be further transformed if they are defined
+// on the patch.
+func (c *Patch) applyCombineFromVariablesPatch(from, to runtime.Object) error { // nolint:gocyclo
+	// Combine patch requires configuration
+	if c.Combine == nil {
+		return errors.Errorf(errRequiredField, "Combine", c.Type)
+	}
+	// Destination field path is required since we can't default to multiple
+	// fields.
+	if c.ToFieldPath == nil {
+		return errors.Errorf(errRequiredField, "ToFieldPath", c.Type)
 	}
 
-	toMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(to)
+	vl := len(c.Combine.Variables)
+
+	if vl < 1 {
```

Comment:
Executive decision made here to error if there isn't at least one variable set. This is validated by a `kubebuilder` comment already so this check is just here for safety, but maybe we want to allow no variables and just do nothing?